File: app/src/main/java/com/example/aichat/SessionSummary.java
package com.example.aichat;

import androidx.room.Ignore;

public class SessionSummary {
    public String sessionId;
    public String title;
    @Ignore
    public String avatar;
    @Ignore
    public String category;
    @Ignore
    public boolean favorite;
    @Ignore
    public boolean pinned;
    @Ignore
    public boolean hidden;
    @Ignore
    public boolean deleted;
    @Ignore
    public String outline;
    public long lastAt;

    public SessionSummary() {}

    @Ignore
    public SessionSummary(String sessionId, String title, long lastAt) {
        this.sessionId = sessionId;
        this.title = title;
        this.lastAt = lastAt;
    }
}

File: app/src/main/java/com/example/aichat/SessionSummary.kt
package com.example.aichat

import androidx.room.Ignore

class SessionSummary {
    @JvmField var sessionId: String = ""
    @JvmField var title: String = ""
    @JvmField var lastAt: Long = 0L
    @JvmField var lastMessage: String = ""

    @Ignore @JvmField var avatar: String = ""
    /** Base64 编码的头像图片：由绑定助手 (assistantId) 的 avatarImageBase64 派生，优先于 avatar 文本。 */
    @Ignore @JvmField var avatarImageBase64: String = ""
    @Ignore @JvmField var category: String = ""
    @Ignore @JvmField var favorite: Boolean = false
    @Ignore @JvmField var pinned: Boolean = false
    @Ignore @JvmField var hidden: Boolean = false
    @Ignore @JvmField var deleted: Boolean = false
    @Ignore @JvmField var outline: String = ""

    constructor()

    @Ignore
    constructor(sessionId: String, title: String, lastAt: Long) {
        this.sessionId = sessionId
        this.title = title
        this.lastAt = lastAt
    }
}
